File: WEEK05/121 주식하기_좋은날/원영.js
/**
 * @param {number[]} prices
 * @return {number}
 */
// 찢었네
var maxProfit = function(prices) {
  let answer = 0;
  let min_day = prices[0];
  for (let i = 1; i < prices.length; i++) {
    if (min_day > prices[i]) {
      min_day = prices[i];
      continue;
    }
    if (min_day < prices[i]) {
      let cul = prices[i] - min_day;
      answer = Math.max(answer, cul);
    }
  }
  
  return answer;
};

File: WEEK05/121 주식하기_좋은날/원영.py
prices = [7,1,5,3,6,4]

min_day = prices[0]
answer = 0
for i in range(1, len(prices)):
  if min_day > prices[i]:
    min_day = prices[i]
    continue

  answer = max(answer, prices[i] - min_day)

print(answer)